### DISCUSSION TITLE
bad data on grids

### DISCUSSION THREAD

**Discussion by FisherFy** [epochpic/epoch#227] [Q&A]
Hello devs:
    When I  use epoch-4.17.1, some errors occur which can be seen in field plots. Specifically, some points may reach infinity. The similar problem is also reported by Zhongwei a year ago when he used epoch-4.17.12. 
![T1dens](https://user-images.githubusercontent.com/38976262/119759097-323df000-beda-11eb-95e1-77a8cc55952c.png)

input.deck, constant.F90 and .../epoch2d/src/deck/deck_subset_block.F90 are attached.
[inputdeck.txt](https://github.com/Warwick-Plasma/epoch/files/6550877/inputdeck.txt)
[constantsF90.txt](https://github.com/Warwick-Plasma/epoch/files/6550800/constantsF90.txt)
[deck_subset_blockF90.txt](https://github.com/Warwick-Plasma/epoch/files/6550802/deck_subset_blockF90.txt)

Additionally, I rerun the case in earlier version v4.10.1, in which  no such problem exists. However, v4.10.1 has a slight different code structure, e.g. the version has no .../epoch2d/src/housekeeping/particle_id_hash.f90 as in later versions. I failed to transplant the modified persist subplot to the v4.10.1(What I had done is that copied modified deck_subset_block.f90 and placed particle_id_hash.f90 in the code). So can you give me some advice on how to make it works?

Thank you very much!

**Comment by TomGoffrey** [comment]
Are you able to try the latest epoch version, or better still, the 4.17-devel branch?

Subsets have had quite a few fixes to them over recent versions, and it's possible the issue you are seeing has already been fixed.

If possible, could you also demonstrate the problem with an unmodified version of EPOCH, or if not possible, provide a patch that works with the latest version? Your `constants.F90` file provided won't compile with the latest version of EPOCH.

**Reply by FisherFy** [reply]
Thanks for your advice!

I turn to the latest version of EPOCH v4.17-devel. However,  some make errors stop the make step. One is that implicit type in SUBROUTINE abort_block in the bottom(line 607) of the ...2d/src/parser/evaluate.F90 ， I just simply commented it out.  The other problem I met is in ..2d/src/housekeeping/setup.F90. 
![1622120373(1)](https://user-images.githubusercontent.com/38976262/119830352-c174f300-bf2e-11eb-8ef0-ee8654fe56c2.png)

Is that a bug?

Both the original un-modified constants.F90 and my modified constants.F90 met the same above problem.

 Thanks again.

**Reply by TomGoffrey** [reply]
Hmm, I have no problem compiling 4.17-devel, and the CI tests all pass.

Did you run `make clean` before compiling? Given this looks like an SDF issue, you should also make sure you ran `git submodule update --recursive` before trying to compile the code.

If neither of those fix the issue, could I check which compiler you're using?

**Reply by FisherFy** [reply]
In fact, I download the zip from git, and copy SDF files from earlier version. If the problem truly caused by SDF issue, I will try the way as recommended.

I have a script for make, as attached. 'Compiler' is in side.
[make_step_sh.txt](https://github.com/Warwick-Plasma/epoch/files/6554530/make_step_sh.txt)

**Reply by TomGoffrey** [reply]
Each version of EPOCH has a specific version of SDF it is intended to work with. I suspect in this case the latest version of EPOCH uses a some new functionality from SDF, which isn't included in the version you copied across.

Downloading the zip from git won't work - as explained in the `README` file. You should either clone recursively, `git clone ... --recursive` or download one of the [releases](https://github.com/Warwick-Plasma/epoch/releases) in `.zip` form, which does include the SDF source code.

**Reply by FisherFy** [reply]
I use the most recent version 4.17-devel on our work station that works well with older versions. But now it comes with make problems at the beginning(As the figure shows). Can you give me some clues how to solve it?
![1622259261(1)](https://user-images.githubusercontent.com/38976262/120057074-05711080-c073-11eb-86e0-2c9b1c9aa9e6.png)

'make' by running this script.
[make_step_sh.txt](https://github.com/Warwick-Plasma/epoch/files/6563995/make_step_sh.txt)

Thanks very much!

**Reply by TomGoffrey** [reply]
Well the first error is that you appear to have tried to load a module which doesn't exist. That's not related to EPOCH.

The remaining errors appear to indicate the presence of Windows/DOS style line ending characters in `src/gen_commit_string.sh`. This suggests you've opened and saved the files using a Windows text editor. You should restore the file to its original form. A tool such as `dos2unix` should fix-up the `\r` issues, although if you have further modified the file, other issues may remain.
